src/node_report.cc

```diff
@@ -1021,6 +1029,106 @@ const static struct {
     }
   }
 #endif
+
+  fprintf(fp, "\nLoaded libraries\n");
+  PrintLoadedLibraries(fp, isolate);
+}
+
+/*******************************************************************************
+ * Functions to print a list of loaded native libraries.
+ *
+ ******************************************************************************/
+#ifdef __linux__
+static int LibraryPrintCallback(struct dl_phdr_info *info, size_t size, void *data) {
+  FILE* fp = (FILE*)data;
+  if (info->dlpi_name != nullptr && *info->dlpi_name != '\0') {
+    fprintf(fp, "  %s\n", info->dlpi_name);
+  }
+  return 0;
+}
+#endif
+
+static void PrintLoadedLibraries(FILE* fp, Isolate* isolate) {
+#ifdef __linux__
+  dl_iterate_phdr(LibraryPrintCallback, fp);
+#elif __APPLE__
+  int i = 0;
+  const char *name = _dyld_get_image_name(i);
+  while (name != nullptr) {
+    fprintf(fp, "  %s\n", name);
+    i++;
+    name = _dyld_get_image_name(i);
+  }
+#elif _AIX
+  // We can't tell in advance how large the buffer needs to be.
+  // Retry until we reach too large a size (1Mb).
+  const unsigned int buffer_inc = 4096;
+  unsigned int buffer_size = buffer_inc;
+  char* buffer = (char*) malloc(buffer_size);
+  int rc = -1;
+  while (buffer != nullptr && rc != 0 && buffer_size < 1024 * 1024) {
+    rc = loadquery(L_GETINFO, buffer, buffer_size);
+    if (rc == 0) {
```

Comment:
I was going to suggest we check for the error that says the buffer was not big enough specifically but since looking at the API doc its the only error we can expect given the parameters we are passing in its ok as is.